### PR TITLE
[Serve] Improve observability for controller restarts

### PR DESCRIPTION
**Pull request by shrekris-anyscale**
## Why are these changes needed?

This change:

* Adds a `Counter` to track the number of times the controller has started
* Logs the time taken to recover deployments.

## Related issue number

Addresses #35667

## Checks

- [X] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - This change relies on existing tests.

### CODE REVIEW

File: python/ray/serve/controller.py [hunk 1]
```diff
@@ -428,6 +430,12 @@ def _create_control_loop_metrics(self):
             {"actor_id": ray.get_runtime_context().get_actor_id()}
         )
 
+    def _count_start(self):
```
Comment (edoakes) [inline python/ray/serve/controller.py:433]: nit: this method is only wrapping a single method call so it's probably unnecessary indirection

**Review by edoakes — APPROVED** [review]

File: python/ray/serve/controller.py (hunk 1 above)
Comment (shrekris-anyscale) [reply python/ray/serve/controller.py]: Makes sense, I removed the indirection.

**Review by GeneDer — APPROVED** [review]

**Review by sihanwang41 — APPROVED** [review]

**Comment by shrekris-anyscale** [comment]
The test failures are unrelated:

* [`test_cli.py`](https://buildkite.com/ray-project/oss-ci-build-pr/builds/31053#0189d68c-391d-42f9-bf90-26a404e30817/2931-3754) has been flaky on `master`:

Screen Shot 2023-08-08 at 2 16 14 PM

* [`Linkcheck`](https://buildkite.com/ray-project/oss-ci-build-pr/builds/31053#0189d68f-5783-477b-8f5f-d246f11ad364/6-4678) has been flaky on `master`:

Screen Shot 2023-08-08 at 2 14 14 PM

* [`test_output`](https://buildkite.com/ray-project/oss-ci-build-pr/builds/31053#0189d681-ba01-4a2b-811c-ce67c8bb2ed7/6987-10390) has been failing on `master`:

Screen Shot 2023-08-08 at 2 15 03 PM

* [`semgrep`](https://buildkite.com/ray-project/oss-ci-build-pr/builds/31053#0189d68c-210f-4c45-bf21-752fdc0a1fdf/503-1678) has been failing on `master`:

Screen Shot 2023-08-08 at 2 15 47 PM

**Comment by shrekris-anyscale** [comment]
@edoakes This change is ready to merge. Could you confirm whether it closes https://github.com/ray-project/ray/issues/35667?

**Comment by edoakes** [comment]
@shrekris-anyscale we should add these to the serve dashboard system tab before closing the issue

**Pull request merged**